### ISSUE TITLE
FEATURE: Add Option to Install to keep Proxy Settings even if seem to be invalid

### ISSUE THREAD

**Issue by gebhard73** [Security-Onion-Solutions/securityonion#3791]
Hi,

for some situations it would be nice to have a third option in the function
whiptail_invalid_proxy() in
https://github.com/Security-Onion-Solutions/securityonion/blob/dev/setup/so-whiptail
to keep the proxy settings even if the settings seem to be invalid because the check has failed.

Reasoning: There are cases where additional manual effort has to be done before a proxy is working, e.g. add a trusted root cert to the system's PKI for intercepting proxies. IMHO it doesn't make sense to cover these edge cases in the script, but it should be possible to use the script for basic configuring even in such cases,

FR: 
- add option "Keep nevertheless" to whiptail_invalid_proxy() if proxy check fails
- rename option "Skip" into "Discard" to continue w/o any proxy settings

See also: https://github.com/Security-Onion-Solutions/securityonion/discussions/3782

Thanks!

**Comment by rwwiv** [comment]
Setup uses the proxy entered, so keeping an invalid configuration is not feasible. That check is in place so that, in cases such as yours, setup will not continue on with the proxy settings given. If additional steps need to occur before a proxy can be used, these should be completed prior to running setup.

I do, however, think the "Skip" option could be renamed to better explain that the installer continues without a proxy if that option is chosen.

**Comment by gebhard73** [comment]
OK, understood. Could you then please document the syntax / settings to add a proxy to the salt config here: https://docs.securityonion.net/en/2.3/proxy.html

I'd volunteer doing so if I get the facts.

**Comment by rwwiv** [comment]
I've updated the docs with more information, both in regards to the salt config as well as other apps' configurations.

**Comment by gebhard73** [comment]
Thanks, this is really helpful. May I ask to add information about providing a user and password for authentication? I guess it's the usual http://USER:PASS@HOST:PORT stuff?

**Comment by gebhard73** [comment]
Look nice, thanks.
